### DISCUSSION TITLE
WinUI 3 and .NET Core 5+ for Xbox

### DISCUSSION THREAD

**Discussion by TylerJaacks** [microsoft/microsoft-ui-xaml#8126] [General]
This is an very stupid and disappointing decision on Microsoft's part but ultimately unexpected

```
From XBox team themselves, you are supposed to use GDK, and plain Win32.

UWP apps and games are community-supported only; partners inside Xbox managed programs (Xbox, Xbox Game Pass, Xbox Game Streaming) should use Microsoft Win32 + GDK.

See https://github.com/microsoft/GDK
```

Your response to people wanting to use the new and supported UWP platform on the Xbox is to get an XDK and GDK is laughable. This defeats the whole purpose of allowing anyone to use there retail Xbox as a development machine. UWP was already limiting enough, now you basically say well f you no more hobbies anymore, shell out more money on our XDKs. I don't want to join the Partner program and release a game. I want to be able to tinker and hobby with my device and build cool things.

**Comment by driver1998** [comment]
Windows.UI.Xaml (System XAML) with .NET 6+ is possible, see https://github.com/gus33000/CsUWP for an example.

But I do wonder if it will run on an Xbox, we knew the barebone .NET 5+ runtime runs on Dev mode just fine, but that is running outside of AppContainer, in an SSH session.

As for WinUI 3 on the Xbox, seriously I would rather wait for the Win32-like environment Chromium Edge uses to go public and let WinUI 3 works on there, hopefully...

Xbox is pretty much the only relevant target of UWP these days, not just for hobbists but also big streaming media compaines. (Imagine building the Netflix or Apple TV app with GDK) If Microsoft really wants to deprecate UWP, the Xbox problem should really be solved.
